docs/functions/generate_name.md

```diff
@@ -0,0 +1,65 @@
+---
+# generated by https://github.com/hashicorp/terraform-plugin-docs
+page_title: "generate_name function - terraform-provider-designation"
+subcategory: ""
+description: |-
+  Generate a name for a convention and with specified parameters
+---
+
+# function: generate_name
+
+Generate a name for a convention and with specified parameters
+
+## Example Usage
+
+```terraform
+data "designation_convention" "this" {
+  definition = "(region)-(stage)-(name)-(random)"
+  variables = [
+    {
+      name = "name"
+    },
+    {
+      name    = "region"
+      default = "we"
+    },
+    {
+      name       = "stage"
+      max_length = 4
+      default    = "dev"
+    },
+    {
+      name       = "random"
+      max_length = 4
+    },
+  ]
+}
+
+resource "random_string" "random" {
+  length           = 16
```

Comment:
der `random`wert in der convention setzt länge auf 4
der `random_string` generiert 16

Schneidet dann designation den Rest ab?

Reply:
ja